src/spectators.h

```diff
@@ -6,46 +6,61 @@
 
 class Creature;
 
-class SpectatorVec
+class Spectators final
 {
 	using Vec = std::vector<Creature*>;
 	using Iterator = Vec::iterator;
 	using ConstIterator = Vec::const_iterator;
 
 public:
-	SpectatorVec() { vec.reserve(32); }
+	Spectators() { spectators.reserve(32); }
 
-	void addSpectators(const SpectatorVec& spectators)
+	void insert(Creature* spectator)
 	{
-		for (Creature* spectator : spectators.vec) {
-			auto it = std::find(vec.begin(), vec.end(), spectator);
-			if (it != end()) {
-				continue;
-			}
-			vec.emplace_back(spectator);
+		auto it = std::find(spectators.begin(), spectators.end(), spectator);
+		if (it == end()) {
+			spectators.emplace_back(spectator);
+		}
+	}
+
+	void insert(Spectators& range)
+	{
+		for (Creature* spectator : range.spectators) {
+			insert(spectator);
+		}
+	}
+	void insert(const Spectators& range)
+	{
+		for (Creature* spectator : range.spectators) {
+			insert(spectator);
 		}
 	}
 
 	void erase(Creature* spectator)
 	{
-		auto it = std::find(vec.begin(), vec.end(), spectator);
+		auto it = std::find(spectators.begin(), spectators.end(), spectator);
 		if (it == end()) {
 			return;
 		}
+
 		std::iter_swap(it, end() - 1);
-		vec.pop_back();
+		spectators.pop_back();
 	}
 
-	size_t size() const { return vec.size(); }
-	bool empty() const { return vec.empty(); }
-	Iterator begin() { return vec.begin(); }
-	ConstIterator begin() const { return vec.begin(); }
-	Iterator end() { return vec.end(); }
-	ConstIterator end() const { return vec.end(); }
-	void emplace_back(Creature* c) { vec.emplace_back(c); }
+	size_t size() const { return spectators.size(); }
+
+	bool empty() const { return spectators.empty(); }
+
+	void emplace_back(Creature* c) { spectators.emplace_back(c); }
+
+	Iterator begin() { return spectators.begin(); }
+	ConstIterator begin() const { return spectators.begin(); }
+
+	Iterator end() { return spectators.end(); }
+	ConstIterator end() const { return spectators.end(); }
```

Comment:
Return `decltype(auto)` from these functions and remove the aliases

Reply:
Thanks!